xarray/tutorial.py

```diff
@@ -42,18 +43,27 @@ def load_dataset(name, cache=True, cache_dir=_default_cache_dir,
 
     """
     longdir = _os.path.expanduser(cache_dir)
+    tmpdir = _os.sep.join((longdir, '.tmp'))
     fullname = name + '.nc'
     localfile = _os.sep.join((longdir, fullname))
+    tmpfile = _os.sep.join((longdir, fullname))
 
     if not _os.path.exists(localfile):
 
         # This will always leave this directory on disk.
         # May want to add an option to remove it.
         if not _os.path.isdir(longdir):
             _os.mkdir(longdir)
+        if not _os.path.isdir(tmpdir):
+            _os.mkdir(tmpdir)
 
         url = '/'.join((github_url, 'raw', 'master', fullname))
-        _urlretrieve(url, localfile)
+        _urlretrieve(url, tmpfile)
+
+        if not _os.path.exists(tmpfile):
+            raise ValueError('File could not be downloaded, please try again')
+
+        _shutil.move(tmpfile, localfile)
```

Comment:
I thought that the issue wasn't that the file isn't downloaded, but rather that it is incompletely downloaded? Will this new temporary step than solve this? 

Is there a way to make ``urlretrieve`` more robust, with checksums or something?